torch/csrc/distributed/c10d/Functional.cpp

```diff
@@ -273,6 +273,31 @@ at::Tensor broadcast(
   return broadcast_(output, src, group_name);
 }
 
+at::Tensor& scatter_(
+    at::Tensor& input,
+    std::vector<at::Tensor> scatter_list,
+    int64_t src,
+    std::string group_name) {
+  c10d::ScatterOptions opts;
+  opts.rootRank = src;
+  std::vector<at::Tensor> outputs{input};
+  std::vector<std::vector<at::Tensor>> inputs{scatter_list};
+
+  auto group = c10d::resolve_process_group(group_name);
+  auto work = group->scatter(outputs, inputs, opts);
```

Comment:
I think we are missing the thunk for `scatter` in https://github.com/pytorch/pytorch/blob/main/torch/csrc/distributed/c10d/PyProcessGroup.hpp. The absence of the thunk prevents C++ from being able to call into process group methods implemented in Python (for `MultiThreadedTest`).

Reply:
Thanks for catching that! Added